h2o-core/src/main/java/hex/FeatureInteractions.java

```diff
@@ -0,0 +1,223 @@
+package hex;
+
+import org.apache.commons.lang.mutable.MutableInt;
+import water.util.TwoDimTable;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+public class FeatureInteractions {
+    
+    private HashMap<String, FeatureInteraction> map;
+
+    public FeatureInteractions() {
+        this.map = new HashMap<>();
+    }
+
+    public static void merge(FeatureInteractions leftFeatureInteractions, FeatureInteractions rightFeatureInteractions) {
+        for (Map.Entry<String,FeatureInteraction> currEntry : rightFeatureInteractions.entrySet()) {
+            if (leftFeatureInteractions.map.containsKey(currEntry.getKey())) {
+                FeatureInteraction leftFeatureInteraction = leftFeatureInteractions.get(currEntry.getKey());
+                FeatureInteraction rightFeatureInteraction = currEntry.getValue();
+                leftFeatureInteraction.gain += rightFeatureInteraction.gain;
+                leftFeatureInteraction.cover += rightFeatureInteraction.cover;
+                leftFeatureInteraction.fScore += rightFeatureInteraction.fScore;
+                leftFeatureInteraction.fScoreWeighted += rightFeatureInteraction.fScoreWeighted;
+                leftFeatureInteraction.averageFScoreWeighted = leftFeatureInteraction.fScoreWeighted / leftFeatureInteraction.fScore;
+                leftFeatureInteraction.averageGain = leftFeatureInteraction.gain / leftFeatureInteraction.fScore;
+                leftFeatureInteraction.expectedGain += rightFeatureInteraction.expectedGain;
+                leftFeatureInteraction.treeIndex += rightFeatureInteraction.treeIndex;
+                leftFeatureInteraction.averageTreeIndex = leftFeatureInteraction.treeIndex / leftFeatureInteraction.fScore;
+                leftFeatureInteraction.treeDepth += rightFeatureInteraction.treeDepth;
+                leftFeatureInteraction.averageTreeDepth = leftFeatureInteraction.treeDepth / leftFeatureInteraction.fScore;
+                leftFeatureInteraction.sumLeafCoversRight += rightFeatureInteraction.sumLeafCoversRight;
+                leftFeatureInteraction.sumLeafCoversLeft += rightFeatureInteraction.sumLeafCoversLeft;
+                leftFeatureInteraction.sumLeafValuesRight += rightFeatureInteraction.sumLeafValuesRight;
+                leftFeatureInteraction.sumLeafValuesLeft += rightFeatureInteraction.sumLeafValuesLeft;
+                leftFeatureInteraction.splitValueHistogram.merge(rightFeatureInteraction.splitValueHistogram);
+            } else {
+                leftFeatureInteractions.put(currEntry.getKey(), currEntry.getValue());
+            }
+        }
+    }
+
+    public int maxDepth() {
+        return Collections.max(this.entrySet(), Comparator.comparingInt(entry -> entry.getValue().depth)).getValue().depth;
+    }
+    
+    public TwoDimTable[] getFeatureInteractions() {
```

Comment:
`FeatureInteractions.getFeatureInteractions` is not very descriptive, what about `getAsTable()`?